capstone/src/main/webapp/multilinevisualization.js

```diff
@@ -12,214 +12,226 @@
+    new Date(2020, 2, 29),
+    new Date(2020, 3, 5),
+    new Date(2020, 3, 12),
+  ];
+  return {
+    'restaurantData': [
+      {
+        'restaurantName': 'Wildfire',
+        'clickData': [2, 10, 10, 8, 3, 18, 20],
+      },
+      {
+        'restaurantName': 'Poke Doke',
+        'clickData': [0, 8, 10, 5, 20, 22, 17],
+      },
+      {
+        'restaurantName': 'The Goog Noodle',
+        'clickData': [20, 18, 12, 21, 20, 19, 15],
+      },
+    ],
+    'dates': dateArr,
+  };
+}
 
-/**
- * Checks the mouse position upon hovering over a graph and displays
- * the closest point to that position. Also greys out all lines but the one with
- * the closest point and displays the name of the restaurant the selected
- * line belongs to
- * @param svg, the graph being hovered over
- * @param path, the path of lines to look through
- */
-function hover(svg, path) {
-  // Setup functions to be called on different actions
-  svg.on('mousemove', moved).on('mouseenter', entered).on('mouseleave', left);
-
-  // Create a blank dot with blank text
-  const dot = svg.append('g').attr('display', 'none');
-  dot.append('circle').attr('r', 2.5);
-  dot.append('text')
-      .attr('font-family', 'sans-serif')
-      .attr('font-size', 10)
-      .attr('text-anchor', 'middle')
-      .attr('y', -8);
-
-  // Function for when the mouse is moved within the graph area
-  function moved() {
-    d3.event.preventDefault();
-    // Get current mouse position
-    const mouse = d3.mouse(this);
-    // The invert function will return the value from the domain given
-    // a value from the range
-    // Range was set when originally declaring x and y to the width/height, so
-    // our mouse values will be properly converted
-    const xm = x.invert(mouse[0]);
-    const ym = y.invert(mouse[1]);
-
-    // The bisectLeft function will return the index in the dates
-    // array you would insert xm in order to maintain a sorted order
-    // Essentially returns an integer index in the array
-    // represented on the x-axis so we know which tick on the x-axis the mouse
-    // is currently closest to
-    // Check both the point to the left and right of current position to see
-    // which is closer
-    const i1 = d3.bisectLeft(data.dates, xm, 1);
-    const i0 = i1 - 1;
-    const i = xm - data.dates[i0] > data.dates[i1] - xm ? i1 : i0;
-
-    // Now look through the different y-values, aka click data for each
-    // restaurant, to find which is closest to the mouse's current y-position
-    // Keep track of the full restaurant object with the closest data point
-    let minTrack = Math.abs(data.restaurantData[0].clickData[i] - ym);
-    let s = data.restaurantData[0];
-    for (let k = 1; k < data.restaurantData.length; k++) {
-      if (Math.abs(data.restaurantData[k].clickData[i] - ym) < minTrack) {
-        minTrack = Math.abs(data.restaurantData[k].clickData[i] - ym);
-        s = data.restaurantData[k];
+function createMultilineVisualization() {
```

Comment:
Consider adding a comment that describes the visualization (what's the x-axis, y-axis, data etc)

Reply:
Great idea, I'll add it before merging